### PR TITLE
Makes guest passes usefull

### PR DESCRIPTION
**Pull request by Robertha89**
the arbitrary limit of 30 minutes seems odd, so lets double it so they can be more useful!
:cl: ppi
tweak: You can now make a guest pass for any duration from 5 too 60 minutes.
/:cl:

### CODE REVIEW

**Comment by Crazylemon64** [comment]
I'm not sure it's the greatest of ideas to let anyone print infinite copies of any ID they have...

**Comment by Robertha89** [comment]
again, each access need to be selected between prints, so it's not the fastest way, but it's more for shit like HoP/Captain is dead, or is demanding paperwork, so you ask for say, robotics access for 3 hours.
the more accesses there are, the more accesses needed to select. i.e too much work to make tons of copies for no raisin

**Comment by KasparoVy** [comment]
Instead of removing the limit outright, why not enable functionality like a department head/captain being able to override the time limit only, otherwise it's still restricted to half an hour?

Otherwise this would just be way too exploitable

**Comment by Fox-McCloud** [comment]
No :-1: 

This produces literally unlimited cards.

I'm fine with refactoring the ID machine to function like TG's and giving each head of staff and ID console, but just being able to spam create guest passes, like no tomorrow?

**Comment by Robertha89** [comment]
You can still make unlimited cards, only issue is you now have to make them 4 times, while here you need to make them once.

There is also the fact that guest pass terminals is very little used for the amount of them there are.

Yes, it's exploitable, but so is other things, allowing it like this gives more power to the people so to speak. 

I mean currently it's at a level where if it was removed outright, it would take weeks for people to notice.

**Comment by TheDZD** [comment]
Being able to print off effectively permanent ID cards with the same access levels that you have is excessive. At the very least with the time limit, you can't stash a guest pass for the entire round in case you lose your ID/get demoted, with this you could very easily toss a guest pass in a personal locker and essentially have permanent insurance in case of demotion.

**Comment by Robertha89** [comment]
But could just as easily give people access without having to go by the hop, there will be abuse, and there will be good use, there is still a time limit that can be set, but you can set it to anything, you have to carry the guest pass in your hand, can't wear it as an id or in pda, so that limits it somewhat.
Just because it can be abused is as much of a valid reason as having black powder in chemistry, it will be abused. the funny thing is, scientists are the only one who does not have access to one, and needs to break in to use it. And really, there is nothing stopping you from stashing your id away, and saying it was stolen, and then get a new one.

**Comment by FalseIncarnate** [comment]
I can support increasing the time limit to 45 minutes or even 1 hour (60 min). You can always print off a new pass if you need extra time for your guest.

But given the purpose of a GUEST pass is for a short-term visit, any long-term granting of access should be left to the heads (specifically HoP) and not any random doc/engi/scientist/officer.

You can safely hand over a guest pass knowing that it will expire so you don't need to track down the visitor later to remove their access. If they don't expire (which I can fairly safely guarantee would be most of the time), then you suddenly are dealing with having to forcibly revoke that access or risk the tide filling your department because some clown (may or may not be THE clown) decided to print off 20+ passes with access to nearly the entire department and leave them in the bar.

TL;DR: longer limit ok, but no more than 1 hour at absolute max. The HoP shouldn't be obsolete because of a guest pass console.

**Comment by Fox-McCloud** [comment]
Even an hour is pretty absurd; that's half the round.

**Comment by Robertha89** [comment]
You set the time, it's not Infinity by default. So if someone asks for one, you set the duration yourself. If you don't want a guy with access for the entire round, the just set it to the amount you want the access to last, you can't set it to infinity because numbers, this proves that you as a player who have been around since before this was a thing, did not know this, which again shows how little they are used.

Anyway, duration changed to max one hour. everything else updated.

**Comment by FalseIncarnate** [comment]
I am quite aware they have you set the time. I was just pointing out people would just set it to an absurd duration like 9999 minutes so they don't need to worry about the person bugging them after a continue round vote potentially passes, effectively rendering the pass infinite duration.

I have used the passes once or twice myself, but they generally are ignored because it's rare that you have someone who needs in and out of a department frequently enough to warrant one but not just get them actual ID access.

**Comment by Fox-McCloud** [comment]
@TheDZD 

Waiting for your final decision.

**Comment by TheDZD** [comment]
1 hour is still fairly long. Honestly I don't see a reason for these to last longer than 30 minutes. They're *guest passes,* not ghetto ID cards.

**Comment by Fox-McCloud** [comment]
just going to go ahead and close this.
